tips/TIP-0018/tip-0018.md

```diff
@@ -0,0 +1,3832 @@
+                        <td>ByteArray[32]</td>
+                        <td>The raw bytes of the <i>NFT ID</i> which is the BLAKE2b-256 hash of the outputID that created it.</td>
+                    </tr>
+                </table>
+            </details>
+        </td>
+    </tr>
+</table>
+
+| :information_source:  Good to know about address format |
+|---------------------------------------------------------|
+
+The <i>Address Type</i> byte of a raw address has an effect on the starting character of the bech32 encoded address,
+which is the recommended address format for user facing applications.
+
+A usual bech32 encoded mainnet address starts with `iota1`, and continues with the bech32 encoded bytes of the address.
+By choosing <i>Address Type</i> as a multiple of 8 for different address types, the first character after the `1`
+separator in the bech32 address will always be different.
+
+| Address | Type Byte as `uint8` | Bech32 Encoded |
+|---------|----------------------|----------------|
+| Ed25519 | 0                    | iota1**q**...  |
+| Alias   | 8                    | iota1**p**...  |
+| NFT     | 16                   | iota1**z**...  |
+
+A user can identify by looking at the address whether it is a signature backed address, a smart contract chain account
+or an NFT address.
+
+##### Storage Deposit Return Unlock Condition
+
+This unlock condition is employed to achieve conditional sending. An output that has
+<i>Storage Deposit Return Unlock Condition</i> specified can only be consumed in a transaction that deposits
+`Return Amount` IOTA coins into `Return Address`. When several of such outputs are consumed, their return amounts per
+`Return Addresses` are summed up and the output side must deposit this total sum per `Return Address`.
+
+###### Additional syntactic transaction validation rule:
+
+- `Minimum Storage Deposit` is the storage deposit in the base currency required for a <i>Basic Output</i> that only
+  has an <i>Address Unlock Condition</i> with `Return Address`, no additional unlock conditions, no features and
+  no native tokens.
+- It must hold true, that `Minimum Storage Deposit` ≤  `Return Amount` ≤ `Amount`.
+
+###### Additional semantic transaction validation rule:
+
+- An output that has <i>Storage Deposit Return Unlock Condition</i> specified must only be consumed and unlocked in a
+  transaction that deposits `Return Amount` IOTA coins to `Return Address` via one or more outputs that:
+   - are of type [<i>Basic Output</i>](#basic-output),
+   - have only an [<i>Address Unlock Condition</i>](#address-unlock-condition) defined, and
+   - have no [<i>Features</i>](#features).
+- When several outputs with <i>Storage Deposit Return Unlock Condition</i> and the same `Return Address` are consumed,
+  their return amounts per `Return Addresses` are summed up and the output side of the transaction must deposit
+  _at least_ this total sum per `Return Address` via output(s) that satisfy the previous condition.
```

Comment:
> We don't allow additional unlock conditions

Ah yes I meant to write this, I agree that you shouldn't have allow unlock conditions.

> Features don't affect the unlocking so we could theoretically allow them.

This is what I think we shouldn't limit

> If you used the SDRUC however, it means you want to send something which is less than the storage deposit, or its a native token. So if the recipient wants to return it with features, they have to pay into the output since features increase the storage deposit.

If the original output (with the SDRUC) had features then it would have contained enough IOTA to cover the storage deposit, hence you can inherently send it back without needing to add any of your own.

If the original output didn't have features then yes you may need to provide more IOTA to cover the storage deposit, but thats up to the sender to decide, I don't think we should limit too much what could be a valid UTXO.

> If you really wanted we could also allow NFTs with only an Address Unlock Condition, but does it make sense?

Yes I think it makes sense to be able to return an NFT it your don't want it. And again its the freedom of the option to do this, as I don't see a technical limitation of allowing it.

> So even if you don't have funds in your wallet, you can satisfy the SDRUC unlock conditions if you send back a basic output with X+Y tokens to return address. (btw in practice I think the receiver would just pocket Y tokens for themselves)

The receiver couldn't pocket the tokens if they couldn't provide the amount to satisfy the unlock.